babel/babel
# @babel/eslint-plugin: Update rules/tests to use @babel/eslint-parser

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | refs #10752, https://github.com/babel/babel/pull/10975
| Patch: Bug Fix?          |
| Major: Breaking Change?  | :+1:
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT


This PR builds on https://github.com/babel/babel/pull/10975 (we should land that first or, if folks prefer, this PR can supersede that one) and does the following:

1. Removes the remaining unnecessary rules from `@babel/eslint-plugin`
1. Uses `@babel/eslint-parser` for `@babel/eslint-plugin`'s tests
1. Start a new private packages called `eslint/babel-eslint-shared-fixtures` so that we can centralize shared fixtures for the ESLint packages
1. Remove rule overrides for stage 4 features. Some of these aren't handled in the ESLint core rules yet (currently blocked by https://github.com/estree/estree/pull/204), but they will be, and so I think it's best to not duplicate throwaway work here.
1. I've deleted all test cases that are not specifically testing the overridden behavior in each rule. I'd like to figure out a way to continue testing core rule tests using these packages, but manually managing copying over tests every release is a huge hassle and not a road I think we should go down. Note that some of these tests cases are very old and contained options that no longer exist.

This should get us back to a pretty sane state that we can use as a foundation to figure out what our maintenance strategy will be here. Excited to start figuring that stuff out!
